src/MockBot.Api/Controllers/ChatController.cs

```diff
@@ -0,0 +1,46 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using MockBot.Api.Interfaces;
+using MockBot.Api.Models;
+
+namespace MockBot.Api.Controllers;
+
+[Route("api/chats")]
+public class ChatController : Controller
+{
+    private readonly IChatService _chatService;
+
+    public ChatController(IChatService chatService)
+    {
+        _chatService = chatService;
+    }
+
+    [HttpGet]
+    public IEnumerable<Chat> FindAll()
+    {
+        return _chatService.FindAll();
+    }
+
+    [HttpGet("{id}")]
+    public Chat Get(Guid id)
+    {
+        return _chatService.Get(id);
+    }
+    
+    [HttpPost]
+    public Chat Post()
+    {
+        return _chatService.Create();
+    }
+    
+    [HttpGet("{chatId}/messages")]
+    public List<Message> GetMessages(Guid chatId)
+    {
+        return _chatService.Get(chatId).Messages;
+    }
+
+    [HttpPost("{chatId}/messages")]
+    public Message PostMessage(Guid chatId, [FromBody] string content)
```

Comment:
The id is autogenerated I think isn't it? I am not sure we can expect the caller to know what the Chat ID is when posting.

Reply:
How else would the message know which chat it is being sent to?

Reply:
I had not realised the distinction between `Chat` and `Message` ... I now understand that this is to add a `Message` to an existing `Chat`, right?

Reply:
That's right